src/app/components/Transactions/TransactionEvents.tsx

```diff
@@ -9,12 +11,15 @@ export const TransactionEvents: FC<{
   addressSwitchOption: AddressSwitchOption
 }> = ({ transaction, addressSwitchOption }) => {
   const { network, layer } = transaction
+  const pagination = useSearchParamsPagination('page')
+  const offset = (pagination.selectedPage - 1) * limit
   if (layer === Layer.consensus) {
     throw AppErrors.UnsupportedLayer
   }
   const eventsQuery = useGetRuntimeEvents(network, layer, {
     tx_hash: transaction.hash,
-    limit: 100, // We want to avoid pagination here, if possible
```

Comment:
Why we want to avoid pagination?

Reply:
Related: https://github.com/oasisprotocol/explorer/issues/226

Reply:
:shrug: I can't find an explanation. 1000 limit was allowed when https://github.com/oasisprotocol/explorer/pull/224 was made